packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -148,14 +180,33 @@ class MediaQueryData {
   /// for subsequent descendants in the widget tree by inserting a new
   /// [MediaQuery] widget using the [MediaQuery.removePadding] factory.
   ///
+  /// Padding is derived from the values of viewInsets and viewPadding.
+  ///
   /// See also:
   ///
-  ///  * [MediaQueryData], which provides some additional detail about this
-  ///    property and how it differs from [viewInsets].
+  ///  * [ui.window], which provides some additional detail about this
+  ///    property and how it relates to [viewInsets] and [viewPadding].
   ///  * [SafeArea], a widget that consumes this padding with a [Padding] widget
   ///    and automatically removes it from the [MediaQuery] for its child.
   final EdgeInsets padding;
 
+  /// The parts of the display that may be obscured by the system UI, but
+  /// respect the padding established below the obstruction.
+  ///
+  /// When a mobile device's keyboard is visible, `viewPadding.bottom`
+  /// corresponds to the bottom of the view beneath it.
```

Comment:
Maybe `This value remains the same regardless of whether the system is is reporting other obstructions in the same physical area of the screen.  For example, a software keyboard on the bottom of the screen that may cover and consume the same area that requires bottom padding will not affect this value.`

I think we should avoid being mobile specific for this doc if possible.